azuredevops/internal/service/memberentitlementmanagement/resource_group_entitlement.go

```diff
@@ -0,0 +1,386 @@
+					Value: struct {
+						AccountLicenseType string `json:"accountLicenseType"`
+						LicensingSource    string `json:"licensingSource"`
+					}{
+						string(*accountLicenseType),
+						licensingSource.(string),
+					},
+				},
+			},
+		})
+
+	if err != nil {
+		return fmt.Errorf("Updating group entitlement: %v", err)
+	}
+
+	result := *patchResponse.Results
+
+	if !*result[0].IsSuccess {
+		return fmt.Errorf("Updating group entitlement: %s", getGroupEntitlementAPIErrorMessage(&result))
+	}
+	return resourceGroupEntitlementRead(d, m)
+}
+
+var uuidRegexp = regexp.MustCompile("^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$")
+
+func importGroupEntitlement(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	_, err := uuid.Parse(d.Id())
+	if err != nil {
+		upn := d.Id()
+		if !uuidRegexp.MatchString(upn) {
+			return nil, fmt.Errorf("Only UUID values can used for import [%s]", upn)
+		}
+		clients := m.(*client.AggregatedClient)
+		result, err := clients.IdentityClient.ReadIdentities(clients.Ctx, identity.ReadIdentitiesArgs{
+			SearchFilter: converter.String("General"),
+			FilterValue:  &upn,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		if result == nil || len(*result) <= 0 {
+			return nil, fmt.Errorf("No entitlement found for [%s]", upn)
+		}
+		if len(*result) > 1 {
+			return nil, fmt.Errorf("More than one entitle found for [%s]", upn)
+		}
+
+		d.SetId((*result)[0].Id.String())
+	}
+	return []*schema.ResourceData{d}, nil
+}
+
+func readGroupEntitlement(clients *client.AggregatedClient, id *uuid.UUID) (*memberentitlementmanagement.GroupEntitlement, error) {
+	return clients.MemberEntitleManagementClient.GetGroupEntitlement(clients.Ctx, memberentitlementmanagement.GetGroupEntitlementArgs{
+		GroupId: id,
+	})
+}
+
+func flattenGroupEntitlement(d *schema.ResourceData, groupEntitlement *memberentitlementmanagement.GroupEntitlement) {
+	d.SetId(groupEntitlement.Id.String())
+	d.Set("descriptor", *groupEntitlement.Group.Descriptor)
+	d.Set("origin", *groupEntitlement.Group.Origin)
+	if groupEntitlement.Group.OriginId != nil {
+		d.Set("origin_id", *groupEntitlement.Group.OriginId)
+	}
+	d.Set("display_name", *groupEntitlement.Group.DisplayName)
+	d.Set("principal_name", *groupEntitlement.Group.PrincipalName)
+	d.Set("account_license_type", string(*groupEntitlement.LicenseRule.AccountLicenseType))
+	d.Set("licensing_source", *groupEntitlement.LicenseRule.LicensingSource)
+}
+
+func expandGroupEntitlement(d *schema.ResourceData) (*memberentitlementmanagement.GroupEntitlement, error) {
+	origin := d.Get("origin").(string)
+	originID := d.Get("origin_id").(string)
+	principalName := d.Get("principal_name").(string)
+	displayName := d.Get("display_name").(string)
+
+	if len(principalName) > 0 && strings.Contains(principalName, "\\") {
+		displayName = strings.Split(principalName, "\\")[1]
+	}
+
+	if len(originID) > 0 && len(principalName) > 0 {
```

Comment:
Constraint has been defined in the schema - `ConflictsWith` and `AtLeastOneOf`,  validation can be removed